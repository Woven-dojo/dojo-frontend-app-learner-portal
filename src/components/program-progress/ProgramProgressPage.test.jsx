import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import AxiosMockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
import { AppContext } from '@edx/frontend-platform/react';

import ProgramProgressPage from './ProgramProgressPage';

const { LMS_BASE_URL } = process.env;

const PROGRAM_UUID = '86eb2e15-1762-498b-aaa2-510ae4ec0ad2';

const APP_CONTEXT = {
  authenticatedUser: {
    userId: 1,
    roles: [],
  },
  config: {
    LMS_BASE_URL,
  },
};

const axiosMock = new AxiosMockAdapter(axios);
axiosMock
  .onGet(`${LMS_BASE_URL}/api/dashboard/v0/programs/${PROGRAM_UUID}/progress_details/`)
  .reply(200, {
    programData: {
      title: 'Program Title',
    },
  });

jest.mock('@edx/frontend-platform/auth', () => ({
  ...jest.requireActual('@edx/frontend-platform/auth'),
  getAuthenticatedHttpClient: jest.fn(),
}));

describe('ProgramProgressPage', () => {
  beforeAll(() => {
    const httpClient = axios.create();
    getAuthenticatedHttpClient.mockImplementation(() => httpClient);
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it('should render the expected HTML', async () => {
    let tree = null;

    await renderer.act(async () => {
      tree = await renderer.create(
        <AppContext.Provider value={APP_CONTEXT}>
          <MemoryRouter
            initialIndex={0}
            initialEntries={[`/test-org/program-progress/${PROGRAM_UUID}`]}
          >
            <div>
              <Route path="/:enterpriseSlug/program-progress/:programUUID">
                <ProgramProgressPage />
              </Route>
            </div>
          </MemoryRouter>
        </AppContext.Provider>,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
