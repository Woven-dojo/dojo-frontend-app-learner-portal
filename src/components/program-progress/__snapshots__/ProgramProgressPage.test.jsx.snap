// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramProgressPage should render the expected HTML 1`] = `
<div>
  <div
    className="col-4 offset-4"
  >
    Program Progress Header Will Go Here
  </div>
  <div
    className="py-5 container-mw-lg container-fluid"
  >
    <div
      className="row"
    >
      <article
        className="col-xs-12 col-lg-12"
      >
        <div>
          Program Courses Will go Here
        </div>
      </article>
    </div>
  </div>
</div>
`;
